monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -71,8 +72,14 @@ def connect(
         self._http_client.additional_headers = {HTTPIslandAPIClient.TOKEN_HEADER_KEY: auth_token}
 
     def _get_authentication_token(self) -> str:
-        response = self._http_client.post("/agent-otp-login", {"otp": self._otp})
-        return response.json()["token"]
+        try:
+            response = self._http_client.post(
+                "/agent-otp-login", {"otp": self._otp.get_secret_value()}
+            )
+            return response.json()["token"]
+        except Exception:
+            # Custom error message to avoid leaking the OTP
+            raise IslandAPIAuthenticationError("Failed to retrieve the authentication token")
```

Comment:
We have no control over what happens in the `HTTPClient`, but we can suppress the error message